shm/fruit.cpp

```diff
@@ -14,7 +13,8 @@ Fruit::~Fruit()
 
 size_t Fruit::getPrice() const
 {
-    return static_cast<size_t>(basePrice_ * ((size_t)(expiry_date_ - time_elapsed_)) / expiry_date_);
+    //return static_cast<size_t>(basePrice_ * ((size_t)(expiry_date_ - time_elapsed_)) / expiry_date_);
```

Comment:
remove commented code. 
This commented code looks good, why do you return basePrice_ instead?